contrib/endpoints/src/api_manager/check_security_rules_test.cc

```diff
@@ -0,0 +1,413 @@
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+//
+
+#include "contrib/endpoints/src/api_manager/check_security_rules.h"
+#include "contrib/endpoints/src/api_manager/context/request_context.h"
+#include "contrib/endpoints/src/api_manager/context/service_context.h"
+#include "contrib/endpoints/src/api_manager/mock_api_manager_environment.h"
+#include "contrib/endpoints/src/api_manager/mock_request.h"
+
+using ::testing::_;
+using ::testing::AllOf;
+using ::testing::Invoke;
+using ::testing::Property;
+using ::testing::Return;
+using ::testing::StrCaseEq;
+using ::testing::StrEq;
+using ::testing::StrNe;
+
+using ::google::api_manager::utils::Status;
+using ::google::protobuf::util::error::Code;
+
+namespace google {
+namespace api_manager {
+
+namespace {
+const char kServiceName[] = R"(
+name: "myfirebaseapp.appspot.com")";
+
+const char kBadServiceName[] = R"(
+name: "badService.appspot.com")";
+
+const char kProducerProjectId[] = R"(
+producer_project_id: "myfirebaseapp"
+)";
+
+const char kApis[] = R"(
+apis {
+  name: "Bookstore"
+  version: "v1"
+  methods {
+    name: "ListShelves"
+    request_type_url: "types.googleapis.com/google.protobuf.Empty"
+    response_type_url: "types.googleapis.com/Bookstore.ListShelvesResponse"
+  }
+  methods {
+    name: "ListBooks"
+    request_type_url: "types.googleapis.com/google.protobuf.Empty"
+    response_type_url: "types.googleapis.com/Bookstore.ListBooksResponse"
+  }
+  methods {
+    name: "CreateBook"
+    request_type_url: "types.googleapis.com/Bookstore.CreateBookRequest"
+    response_type_url: "types.googleapis.com/Bookstore.Book"
+  }
+})";
+
+const char kAuthentication[] = R"(
+authentication {
+  providers {
+    id: "issuer1"
+    issuer: "https://issuer1.com"
+  }
+  providers {
+    id: "issuer2"
+    issuer: "https://issuer2.com"
+    jwks_uri: "https://issuer2.com/pubkey"
+  }
+  rules {
+    selector: "ListShelves"
+    requirements {
+      provider_id: "issuer1"
+    }
+    requirements {
+      provider_id: "issuer2"
+    }
+  }
+})";
+
+const char kHttp[] = R"(
+http {
+  rules {
+    selector: "ListShelves"
+    get: "/ListShelves"
+  }
+}
+control {
+  environment: "http://127.0.0.1:8081"
+})";
+
+const char kJwtEmailPayload[] =
+    R"({"iss":"https://accounts.google.com","iat":1486575396,"exp":1486578996,"aud":"https://myfirebaseapp.appspot.com","sub":"113424383671131376652","email_verified":true,"azp":"limin-429@appspot.gserviceaccount.com","email":"limin-429@appspot.gserviceaccount.com"})";
```

Comment:
Same here, can we make this consistent with everything else in this file by merging this line to the end of previous one?